pilot/pkg/bootstrap/istio_ca.go

```diff
@@ -429,18 +431,28 @@ func (s *Server) createIstioCA(opts *caOptions) (*ca.IstioCA, error) {
 	var istioGenerated bool
 	var err error
 
+	// For mounted certificates (if any) determine the set of files.
+	// If not found - we'll load from K8S Secret.
 	fileBundle, err := detectSigningCABundle()
 	if err != nil {
 		return nil, fmt.Errorf("unable to determine signing file format %v", err)
 	}
+	// the fileBundle will have file paths - but they may not exist.
+	// TODO: if no file is mounted we may fallback to loading from secrets.
+
 	if _, err := os.Stat(fileBundle.SigningKeyFile); err == nil {
 		detectedSigningCABundle = true
 		if _, err := os.Stat(path.Join(LocalCertDir.Get(), ca.IstioGenerated)); err == nil {
 			istioGenerated = true
 		}
 	}
+	// TODO: if the cert chain length > 1 - we can't refresh so we can use the other
+	// code path, and ignore detectedSigningCABundle file (which really triggers issuing
+	// refresh certs)
 
 	if !detectedSigningCABundle || (features.UseCacertsForSelfSignedCA && istioGenerated) {
+		// No mounted certs or mounted, but istio-generated.
```

Comment:
No mounted certs found

This maybe confusing, because istio generated secret is now also mounted

Reply:
I didn't find that - you mean istio-ca-secret is mounted ? Why ? 

If you mean that we can no generate cacerts (the one that is mounted) - I think we should discuss if we really need that. It was added very recently and we can probably still remove that option and simplify.

That means: either 'simple' istio self generated, no fuss. Or externally generated.

Reply:
Could you clarify this statement: "If you mean that we can no generate cacerts (the one that is mounted) - I think we should discuss if we really need that. It was added very recently and we can probably still remove that option and simplify."?

The reasoning behind this change was to merge `istio-ca-secret` with `cacerts` so the root certificate has consistent naming regardless of whether or not it is istio generated or plugged in.